tests/test_page_inputs.py

```diff
@@ -117,3 +117,30 @@ def test_http_response_headers_init_invalid():
     with pytest.raises(TypeError):
         response = HttpResponse("http://example.com", body=b"",
                                 headers=123)
+
+
+def test_http_response_selectors(book_list_html_response):
+    title = "All products | Books to Scrape - Sandbox"
+
+    assert title == book_list_html_response.css("title ::text").get("").strip()
+    assert title == book_list_html_response.xpath("//title/text()").get("").strip()
+
+
+def test_http_response_json():
+    http_body = HttpResponseBody(b'{"key": "value"}')
+    response = HttpResponse("http://example.com", body=http_body)
+
+    assert response.json() == {"key": "value"}
```

Comment:
For response.json we may check a couple more things, similar to test_http_response_body_json

Reply:
Good call. Added them in https://github.com/scrapinghub/web-poet/pull/31/commits/3aed4915fc1c44dee8c15349a7055b5c25ebd197.